File: src/client/FantomasTools/Trivia/Encoders.fs
module FantomasTools.Client.Trivia.Encoders

open Thoth.Json
open TriviaViewer.Shared
open FantomasTools.Client.Trivia.Model

let encodeParseRequest (pr: ParseRequest) =
    Encode.object [ "sourceCode", Encode.string pr.SourceCode
                    "defines", List.map Encode.string pr.Defines |> Encode.list
                    "fileName", Encode.string pr.FileName ]
    |> Encode.toString 4

let encodeUrlModel code (model: Model) =
    Encode.object [ "code", Encode.string code // the "code" key is a convention
                    "defines", Encode.string model.Defines
                    "isFsi", Encode.bool model.IsFsi ]

File: _dist_/bin/Trivia/Encoders.js
import { list, object, toString } from "../.fable/Thoth.Json.5.0.0/Encode.fs.js";
import { map } from "../.fable/fable-library.3.1.1/List.js";

export function encodeParseRequest(pr) {
    return toString(4, object([["sourceCode", pr.SourceCode], ["defines", list(map((value_1) => value_1, pr.Defines))], ["fileName", pr.FileName]]));
}

export function encodeUrlModel(code, model) {
    return object([["code", code], ["defines", model.Defines], ["isFsi", model.IsFsi]]);
}

